pkg/monitortestlibrary/disruptionlibrary/collect.go

```diff
@@ -0,0 +1,86 @@
+package disruptionlibrary
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	monitorserialization "github.com/openshift/origin/pkg/monitor/serialization"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+)
+
+func CollectIntervalsForPods(ctx context.Context, kubeClient kubernetes.Interface, namespace string, labelSelector labels.Selector) (monitorapi.Intervals, []*junitapi.JUnitTestCase, []error) {
```

Comment:
collection moved from the existing network logs collection.  Might be worth parallelizing now.